src/mo_frontend/typing.ml

```diff
@@ -2137,6 +2112,36 @@ and check_exp_field env (ef : exp_field) fts =
   | None ->
     ignore (infer_exp env exp)
 
+and check_func_step ?(sub=sub) in_actor env (shared_pat, pat, typ_opt, exp) (s, c, ts1, ts2) : (env * T.typ) =
```

Comment:
redefining sub at the call-sites feels a bit hacky? Is there a better way to factor this?

Reply:
done, thanks! It turned out better than before!